app/src/main/java/com/amaze/filemanager/ui/dialogs/ColorPickerDialog.java

```diff
@@ -311,29 +304,9 @@ public ColorItemPair(Integer first, int[] second) {
     void onAcceptedConfig();
   }
 
-  @Override
-  protected Parcelable onSaveInstanceState() {
-    final SavedState myState = new SavedState(super.onSaveInstanceState());
-    myState.selectedItem = selectedIndex;
```

Comment:
How is instance saved?

Reply:
It crashes on rotation:
```
## Issue explanation (write below this line)



## Exception
* __User Action:__ UI Error
* __Request:__ Application crash
* __Version:__ 3.6.0
* __OS:__ Linux Android 11 - 30
* __Device:__ emulator64_x86_64_arm64
* __Model:__ sdk_gphone64_x86_64
* __Product:__ sdk_gphone64_x86_64
<details><summary><b>Crash log </b></summary><p>

```
java.lang.NullPointerException: Attempt to invoke virtual method 'int com.amaze.filemanager.ui.colors.UserColorPreferences.getAccent()' on a null object reference
	at com.amaze.filemanager.ui.dialogs.ColorPickerDialog.onBindDialogView(ColorPickerDialog.java:134)
	at androidx.preference.PreferenceDialogFragmentCompat.onCreateDialog(PreferenceDialogFragmentCompat.java:148)
	at com.amaze.filemanager.ui.dialogs.ColorPickerDialog.onCreateDialog(ColorPickerDialog.java:245)
	at androidx.fragment.app.DialogFragment.onGetLayoutInflater(DialogFragment.java:419)
	at androidx.fragment.app.Fragment.performGetLayoutInflater(Fragment.java:1484)
	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:320)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1187)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1356)
	at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1434)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1497)
	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2625)
	at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:2577)
	at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:247)
	at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:541)
	at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:210)
	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1454)
	at android.app.Activity.performStart(Activity.java:7946)
	at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3558)
	at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
	at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2138)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7858)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:984)

```
</details>
<hr>

```

Reply:
Ahhh... this is a hard one, since they have rearranged method placements at `PreferenceDialogFragmentCompat` and `DialogPreference`.

Need to find what is backing `ColorPickerDialog` or make one if there is not - I could make the properties set in `ColorPickerDialog.setColorPreference()` to be found somewhere else, but not the listener.

Reply:
7b10865 should fix this. Perhaps the ugliest method (save params to Bundle as arguments to ColorPickerDialog), but seems no other way can do currently.

Selected index at ColorPickerDialog is not persisted across screen rotates, but current method to layout options at the dialog makes difficult to programmatically select the option. Any ideas?

Reply:
> Selected index at ColorPickerDialog is not persisted across screen rotates, but current method to layout options at the dialog makes difficult to programmatically select the option. Any ideas?

I remember this was a bit issue when I originally created the dialog. `select(selectedItem, true);` seems to have been the solution.

Reply:
It seems that the assumption that there is only one dialog at a time might be wrong.

Reply:
The issue seems to be very weird, selecting the last option and rotating the device selects everything.

Reply:
Perhaps [this](https://stackoverflow.com/a/28108909) may help. 10b3380.

Reply:
Only fixes for rotation, not for any other action that requires a saved state, like pausing and recreating the activity, this will have to be an issue that is fixed later though, for now it is good enough.